### PR TITLE
chore: move CI to CircleCI

### PR DESCRIPTION
**Pull request by andipaetzold**
- Moving to CircleCI
- Adds new step to check whether `npm run build` was executed
- run `npm run build`

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -99698,7 +99698,7 @@ module.exports = JSON.parse('{"dots":{"interval":80,"frames":["⠋","⠙","⠹",
...
...n":"16.0.3","description":"Loads environment variables from .env file","main":"lib/main.js","types":"lib/main.d.ts","exports":{".":{"require":"./lib/main.js","types":"./lib/main.d.ts","default":"./lib/main.js"},"./config":"./config.js","./config.js":"./config.js","./lib/env-options":"./lib/env-options.js","./lib/env-options.js":"./lib/env-options.js","./lib/cli-options":"./lib/cli-options.js","./lib/cli-options.js":"./lib/cli-options.js","./package.json":"./package.json"},"scripts":{"dts-check":"tsc --project tests/types/tsconfig.json","lint":"standard","lint-readme":"standard-markdown","pretest":"npm run lint && npm run dts-check","test":"tap tests/*.js --100 -Rspec","prerelease":"npm test","release":"standard-version"},"repository":{"type":"git","url":"git://github.com/motdotla/dotenv.git"},"keywords":["dotenv","env",".env","environment","variables","config","settings"],"readmeFilename":"README.md","license":"BSD-2-Clause","devDependencies":{"@types/node":"^17.0.9","decache":"^4.6.1","dtslint":"^3.7.0","sinon":"^12.0.1","standard":"^16.0.4","standard-markdown":"^7.1.0","standard-version":"^9.3.2","tap":"^15.1.6","tar":"^6.1.11","typescript":"^4.5.4"},"engines":{"node":">=12"}}');
```
Comment (andipaetzold) [inline dist/index.js:99701]: Great that package.jsons end up in the final build....

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - node/install
+
+      - run: npm ci
+      - run: npm run format-check
+      - run: npm run lint
+      - run: npm run build
+      - run:
+          name: Build output is up-to-date
+          command: |
+            if ! [[ -z "$(git status --porcelain)" ]]; then
```
Comment (vikaskumr) [inline .circleci/config.yml:23]: Just one question may be a noob question, what does this command check?

**Review by vikaskumr — APPROVED** [review]

File: dist/index.js (hunk 1 above)
Comment (vikaskumr) [reply dist/index.js:99701]: Did you ran `npm run build` or anything else, just wondering how my local files got in the final build?

File: .circleci/config.yml (hunk 2 above)
Comment (andipaetzold) [reply .circleci/config.yml:23]: If there are uncommited files in the repo, the CI job is exited with status code `1` (error). That way we ensure the `npm run build` step does not cause any changes and the `dist` folder is already up-to-date.

`git status --porcelain` is a more script-friendly version of `git status`

File: dist/index.js (hunk 1 above)
Comment (andipaetzold) [reply dist/index.js:99701]: Just ran `npm run build`

**Pull request merged**